test/perf-tests.js

```diff
@@ -48,26 +63,103 @@ describe('Licode Performance', function() {
+      if (spineNum < numSpinesWithAdditionalParticipant) {
+        numParticipantsInSpine += 1;
+      }
+      return numParticipantsInSpine;
+    }
+
+    let totalSpines = numRooms * spinesPerRoom;
+    let timeToWaitAfterLastSpine = 3 * 60 * 1000;
+
+    for (let classId = 0; classId < numRooms; classId += 1) {
+      for (let spineId = 0; spineId < spinesPerRoom; spineId += 1) {
+        let numPubsInSpine = getNumParticipants(spineId, numPubs);
+        let numSubsInSpine = getNumParticipants(spineId, numSubs);
+        const totalStreamsPerSpine = numPubsInSpine + numPubs * numSubsInSpine;
+        const totalStats = totalStreamsPerSpine * 3;
+
+        let customTest = Object.assign({}, test);
+        customTest.id = classId;
+        customTest.testId += '_' + spineId;
+        customTest.basicExampleUrl += '?room=' + customTest.id;
+        customTest.numPublishers = numPubsInSpine;
+        customTest.numSubscribers = numSubsInSpine;
+        let result = {};
+        let currentSpine = classId * spinesPerRoom + spineId;
+        promises.push(
+            getSpineInstanceFromClass(classId).wait(currentSpine * 30 * 1000)
+            .then(() => getSpineInstanceFromClass(classId).runTest(customTest))
+            .then(() => getSpineInstanceFromClass(classId).wait(timeToWaitAfterLastSpine +
+                                                                (totalSpines - currentSpine) * 30 * 1000))
+            .then(() => getSpineInstanceFromClass(classId).getResult(customTest))
+            .then(testResult => {
+              result = testResult;
+              return getSpineInstanceFromClass(classId).stopTest(customTest);
+            })
+            .then(() => {
+              should.exist(result);
+              should.exist(result.stats);
+              should.exist(result.alive);
+              should.exist(result.stats.bitrateCalculated);
+              should.exist(result.alive.up);
+
+              log(result.alive);
+
+              result.alive.up.should.be.above(Math.floor(totalStreamsPerSpine * 0.8));
+              result.alive.down.should.be.below(Math.ceil(totalStreamsPerSpine * 0.1));
+              result.stats.bitrateCalculated.length.should.be.above(Math.floor(totalStats * 0.8));
+
+              for (let index = 2; index < result.stats.bitrateCalculated.length; index += 3) {
+                result.stats.bitrateCalculated[index].should.be.above(MAX_VIDEO_BITRATE * 0.7);
+              }
+            }));
+        currentSpine++;
+      }
+    }
+
+    return Promise.all(promises);
+  }
+
+  beforeEach(function() {
+    log('Starting:', this.currentTest.title);
+  });
+
+  afterEach(function() {
+    log('Finished:', this.currentTest.title, '[' + this.currentTest.state + ']');
+    testId += 1;
   });
 
-  afterEach(() => {
-    return getLicodeInstance().stopLicode();
+  describe('Default Licode Configuration', function() {
+    beforeEach(function() {
+      test = Object.assign({}, BASIC_TEST);
```

Comment:
For readability I'd consider passing the test configuration (`BASIC_TEST`) as a parameter in `runTestWith`